packages/jupyter-ai/jupyter_ai/reply_processor.py

```diff
@@ -0,0 +1,29 @@
+import asyncio
+from typing import Dict
+from jupyter_ai.handlers import ChatHandler
+from ray.util.queue import Queue
+
+
+class ReplyProcessor():
+    """A single processor to distribute replies"""
+
+    def __init__(self, handlers: Dict[str, ChatHandler], queue: Queue, log):
+        self.handlers = handlers
+        self.queue = queue
+        self.log = log
+
+    def process(self, message):
+        self.log.debug('Processing message %s in ReplyProcessor', message)
+        for handler in self.handlers.values():
+            if not handler:
+                continue
+            
+            handler.broadcast_message(message)
+            break
+
+    async def start(self):
+        while True:
+            if not self.queue.empty():
+                self.process(self.queue.get())
+            
+            await asyncio.sleep(5)
```

Comment:
We should test, but I believe if you await on the queue's get method, it will async await for the next item and process it instantly. The only reason we would need the sleep here is to throttle the replies, which I am not sure it needed.

Reply:
I believe there was something regarding the order in async version.

Reply:
Yeah, have to test this with await added.

Reply:
The server hangs if I remove sleep, added the async get for queue.